addons/stock/models/stock_picking.py

```diff
@@ -680,6 +680,17 @@ def action_assign(self):
         package_level_done.write({'is_done': False})
         moves._action_assign()
         package_level_done.write({'is_done': True})
+
+        # run scheduler for moves are not fully reserved
+        for move in moves.filtered(lambda m: m.state in ['confirmed', 'partially_available']):
+            # get all the parents of the move's source location
+            cur_location = move.location_id
+            parent_locations = cur_location
+            while cur_location.location_id:
+                cur_location = cur_location.location_id
+                parent_locations |= cur_location
+            move.product_id.orderpoint_ids.filtered(lambda op: op.location_id in parent_locations and op.trigger == 'auto')._procure_orderpoint_confirm(company_id=move.company_id, raise_user_error=False)
```

Comment:
performance issue here

there's two things you should do
1) call _procure_orderpoint_confirm once for all the orderpoint linked to a product
2) get rid of the while .. going up with the parent_id column and use a child_of/parent_of operator

Reply:
ideally we would a have a data structure containing somehow the grouped products, locations and companies impacted (maybe we could infer the company from the location thoguh)

something similar to this (but the grouping won't work)
``` python
res = defaultdict(lambda: self.env['stock.location'], self.env['res.company'])
for move in moves:
    res[move.product_id] = (move.location_id, move.company_id)
    
for product, (location, company) in res.iteritems():
    orderpoints = self.env['stock.warehouse.orderpoint'].search([
        ('product_id', '=', product.id),
        ('trigger', '=', 'auto'),
        ('location_id', 'parent_of', location.id)
    ])
    orderpoints._procure_orderpoint_confirm(company_id=company, raise_user_error=False)
```
we could have the same orderpoints triggered two times if two stock moves are processed in wh/stock/shelf1 and wh/stock/shelf2, mybe we could in a second times merge these two liens by a sigle wh/stoch

Reply:
limit=1 on search
store the found orderpoint in a recordset
call _procure_orderpoint_confirm in all orderpoints at once

Reply:
``` python
orderpoints_by_company = defaultdict(lambda: self.env['stock.location.orderpoint'])
for move in moves:
    orderpoints = self.env['stock.warehouse.orderpoint'].search([
        ('product_id', '=', move.product_id.id),
        ('trigger', '=', 'auto'),
        ('location_id', 'parent_of', move.location_id.id),
        ('company_id', '=', move.company_id.id)
    ], limit=1)
    if orderpoints:
        orderpoints_by_company[move.company_id] |= orderpoints.id
for company, orderpoints in orderpoints_by_company.items():
    orderpoints._procure_orderpoint_confirm(company_id=company, raise_user_error=False)
```

Reply:
@amoyaux I think we can remove ('company_id', '=', move.company_id.id) in the search since we already have the location.

Reply:
@sle-odoo @amoyaux Change applied